client/result/result.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```

Comment:
Awesome (thumbsup)!

Is `Options` the only thing shared? If so, might make more sense to leave that in `bootstrap/result` too so bootstrap results and bootstrap result options don't end up living in separate packages